### PR TITLE
Denote stale glucose (LOOP-927)

### PR DESCRIPTION
**Pull request by ps2**
Update based on design review of DIY merge. https://tidepool.atlassian.net/browse/LOOP-927

### CODE REVIEW

**Comment by ps2** [comment]
* A single place for specification of glucose age time interval (LoopSettings?). Currently hardcoded at 15 minutes.
* HUDInterfaceController in WatchApp Extension should not display old bg data (second page of main watch app)
* Complications updated to show '---' when bg data is missing. 
* Complications should register future timeline entries to allow bg data to go stale without updates from Loop.

**Comment by ps2** [comment]
Watch app and complications are now also showing Loop completion freshness using color. BG going stale is still denoted with dashes.

File: LoopCore/LoopSettings.swift [hunk 1]
```diff
@@ -38,6 +38,11 @@ public struct LoopSettings: Equatable {
 
     /// The amount of time since a given date that data should be considered valid
     public let recencyInterval = TimeInterval(minutes: 15)
+    
+    /// Loop completion aging category limits
+    public let completionFreshLimit = TimeInterval(minutes: 6)
+    public let completionAgingLimit = TimeInterval(minutes: 16)
+    public let completionStaleLimit = TimeInterval(hours: 12)
```
Comment (nhamming) [inline LoopCore/LoopSettings.swift:45]: Should these be used above in `LoopCompletionFreshness`? 

Also, I thought the PR comment said `stale` was hardcoded to 15 mins (`recencyInterval`), but `aging` is allowed to be a max of 16 minutes. I think I'm missing something.

File: WatchApp Extension/Controllers/HUDInterfaceController.swift [hunk 2]
```diff
@@ -48,9 +50,10 @@ class HUDInterfaceController: WKInterfaceController {
             return
         }
 
-        glucoseLabel.setHidden(true)
+        glucoseLabel.setText("---")
+        glucoseLabel.setHidden(false)
```
Comment (nhamming) [inline WatchApp Extension/Controllers/HUDInterfaceController.swift:54]: Given this line, Line 62 isn't needed.

File: WatchApp Extension/Extensions/CLKComplicationTemplate.swift [hunk 3]
```diff
@@ -10,37 +10,83 @@ import ClockKit
...
+        
+        let tintColor: UIColor
+        
+        switch loopCompletionFreshness {
+        case .fresh:
+            tintColor = .tintColor
+        case .aging:
+            tintColor = .agingColor
+        case .stale:
+            tintColor = .staleColor
```
Comment (nhamming) [inline WatchApp Extension/Extensions/CLKComplicationTemplate.swift:85]: I'm curious why you don't use the `LoopCompletionFreshness` more often when determining the glucose freshness?

Also, isn't it possible that the glucose is considered `stale` using `recencyInterval`, but gets the style of `aging`? I think I'm missing the different between what the `recencyInterval` calls `stale` and what the `LoopCompletionFreshness` calls `stale`.

**Review by nhamming — APPROVED** [review]
A question about why there seems to be 2 different definitions of `stale`. Probably me missing something.

**Review by darinkrauss — APPROVED** [review]
LGTM. More only questions were around the use of LoopCompletionFreshness versus RecencyInterval, but @nhamming covered that. 🚀

File: LoopCore/LoopSettings.swift (hunk 1 above)
Comment (ps2) [reply LoopCore/LoopSettings.swift:45]: There are two different kinds of staleness being addressed here. This ticket started out with just addressing glucose (on the phone).  Digging into how staleness is shown on the watch, to ensure consistency, also brought some issues with loop completion staleness that needed to be addressed.  Those two things can age independently.  You can do a fresh loop off of < 15m old glucose data, and you can also have fresh glucose data, but a loop that hasn't completed in > 16m due to other errors.

File: WatchApp Extension/Extensions/CLKComplicationTemplate.swift (hunk 3 above)
Comment (ps2) [reply WatchApp Extension/Extensions/CLKComplicationTemplate.swift:85]: Hopefully what I explained above explains this, but let me know if you still have questions.

**Comment by ps2** [comment]
Re-requested reviews, just to make sure my responses address your comments before merging.

**Review by darinkrauss — APPROVED** [review]
LGTM! 🚀 

If `RecencyInterval` is specific to glucose values, then perhaps renaming this to something more specific might clarify the difference.

**Comment by ps2** [comment]
@darinkrauss It's not just for glucose data. It's also for pump data.  It's about the age limit of data Loop is willing to use for creating a forecast. Maybe `inputDataRecencyInterval`?

File: LoopCore/LoopSettings.swift (hunk 1 above)
Comment (nhamming) [reply LoopCore/LoopSettings.swift:45]: I see. Thanks for clarifying.

**Review by nhamming — APPROVED** [review]
LGTM

**Comment by darinkrauss** [comment]
@ps2 Yes, something to distinguish it from the loop completion freshness. But, you can also do this in a later code change.

**Review by darinkrauss — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**